### PR TITLE
Allow the clones to be specified by an array in the event

### PR DESCRIPTION
**Pull request by alcanzar**
This change allows the list of clones to come from a field in the event for those cases where you don't know the number of clones that will be needed ahead of time.

### CODE REVIEW

**Comment by jordansissel** [comment]
I think I understand the goal of this change, but I am not sure of the use case. Can you help me understand better how you would use this?

**Comment by alcanzar** [comment]
This came up on stack overflow: http://stackoverflow.com/questions/42447294/how-do-you-split-json-into-multiple-events-in-logstash-where-events-are-parts-o/42448010#42448010

The person had a weird structure where they had some common information at the top level and then some events that were related and wanted to dynamically create clones with names of the specific events and then remove the non-relevant parts.

Basically if you don't know ahead of time how many clones you need until you inspect the data, the use case is that you build up that list in a field (or the field is provided as part of the event), and then you dynamically create the number of clones you need so that you can get enough replicates to insert the exploded events.
